src/components/surveys/survey-design/question-edit/QuestionEditRhs.tsx

```diff
@@ -133,7 +133,8 @@ const QuestionEditRhs: FunctionComponent<QuestionEditProps> = ({step, onChange,
               step: {...step},
               onChange: onChange,
               q_type: getQuestionId(step),
-            }}></Factory>
+            }}
+            key={step.identifier}></Factory>
```

Comment:
Specify a key, so that child components are re-rendered when switching between steps of the same question type.

Before -- the inputs match the last question, rather than retaining the original inputs for that step.

https://github.com/BridgeDigitalHealth/open-bridge-web/assets/26949006/8dba45ac-6100-4e74-8acb-73e7577bdf77

After -- unique inputs per step are retained when switching between steps.

https://github.com/BridgeDigitalHealth/open-bridge-web/assets/26949006/c59df58d-c84e-48d3-971a-db0800fa8413

Reply:
Seems like the real issue is probably that the `Year` component doesn't update when it gets a new `step` prop? It looks like `Year` is neither fully controlled or fully uncontrolled--you probably want to make it fully controlled. This is an ok stopgap if that isn't a simple change (or if some of the other `Factory` components have the same bug)

Reply:
If you are using the step.identifier as the key, then what happens when you change the step identifier?

Reply:
@nickgros: agreed! For now, I've added TODO comments to make Year and Time fully controlled components.

@syoung-smallwisdom: when the step identifier is updated in `QuestionEditRhs`, the `survey` state (which includes the current `step`) is updated in `SurveyDesign` via a callback. Since the state was updated, `SurveyDesign` is re-rendered and the updated `step` is passed as a prop to `QuestionEditRhs` and then to `Factory`. Since `step` has a new identifier and thus a new key, the state within the `Factory` component is [reset](https://react.dev/reference/react/useState#resetting-state-with-a-key). 

For `Year`, the state will be initialized using the current values of `step`, which will match the last valid configuration for the Year. 

Last state was valid:

https://github.com/BridgeDigitalHealth/open-bridge-web/assets/26949006/863aa7e8-de32-4b69-8bfd-67946752257f

Last state was invalid:

https://github.com/BridgeDigitalHealth/open-bridge-web/assets/26949006/49522563-5a00-44ce-82d5-fb396af25642

Reply:
Thanks!